### DISCUSSION TITLE
riscv32-zephyr-elf  don't support it anymore？

### DISCUSSION THREAD

**Discussion by ilyhcg** [zephyrproject-rtos/sdk-ng#624] [Q&A] [answered]
riscv32-zephyr-elf  don't support it anymore？Currently the hardware is 32-bit and has no compiler

**Comment by keith-packard** [comment, marked as answer]
32-bit risc-v uses the same toolchain as 64-bit risc-v. It's called riscv64-zephyr-elf because that toolchain builds for 64-bit targets by default.  You specify `-march=` and `-mabi=` flags to tell the tools which architecture you're using:
```
riscv64-zephyr-elf-gcc -march=rv32imac -mabi=ilp32 -c hello-world.c
```
builds for a typical 32-bit integer-only target.

**Reply by ilyhcg** [reply]
got it.thanks
